src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Iterator, List, Union, Dict
+
+import pandas as pd
+
+from gluonts.dataset.common import Dataset, DataEntry, ProcessDataEntry
+from gluonts.dataset.field_names import FieldName
+
+
+@dataclass
+class DataFramesDataset(Dataset):
+    """
+    A pd.DataFrame-based dataset type.
+
+    This class is constructed with a collection of pd.DataFrame-objects where each
+    DataFrame is representing one time series. A target and a timestamp columns are
+    essential. Furthermore, static/dynamic real/categorical features can be specified.
+
+    Parameters
+    ----------
+    dataframes: Union[List[pd.DataFrame], Dict[str, pd.DataFrame]]
+        List or Dict of 'pd.DataFrame's containing at least timestamp and target columns.
+        If a Dict is provided, the key will be the associated 'item_id'.
+    target: str or List[str]
+        Name of the column that contains the target time series.
+        For multivariate targets `target` is expecting a list of column names.
+    timestamp: str
+        Name of the column that contains the timestamp information.
+    freq: str
+        Frequency of observations in the time series. Must be a valid pandas frequency.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
```

Comment:
I hate this -- but I don't know a better solution at the moment.

We have hard-coded these field-names, but it would be great if we could support arbitrary fields.

One way to solve this could be to take a dict of fields instead

```py
DataSet(
    features={"feat_dynamic_real": [...])
)
```